core/frontend/src/DisplayStyleState.ts

```diff
@@ -437,8 +438,8 @@ export abstract class DisplayStyleState extends ElementState implements DisplayS
   }
 
   /** @internal */
-  public findMapLayerIndexByNameAndUrl(name: string, url: string, isOverlay: boolean) {
-    return this.getMapLayers(isOverlay).findIndex((x) => x.matchesNameAndUrl(name, url));
+  public findMapLayerIndexByNameAndSource(name: string, source: string, isOverlay: boolean) {
+    return this.getMapLayers(isOverlay).findIndex((layer) => layer instanceof ImageMapLayerSettings && layer.matchesNameAndSource(name, source));
```

Comment:
@RBBentley I think `layer instanceof ImageMapLayerSettings` should be removed here

Reply:
Done...